source/geod24/LocalRest.d

```diff
@@ -820,6 +823,24 @@ public final class RemoteAPI (API, alias S = VibeJSONSerializer!()) : API
         {
             C.send(this.childTid, FilterAPI(""));
         }
+
+        /***********************************************************************
+
+            Call the provided delegate with a scoped copy of this RemoteAPI
+            with a custom timeout.
+
+            Params:
+                timeout = the new timeout to use
+                dg = the delegate to call with the new scoped RemoteAPI copy
+
+        ***********************************************************************/
+
+        public void withTimeout (Dg) (Duration timeout, Dg dg) @trusted
+            if (is(typeof({ dg(RemoteAPI.init); })))
```

Comment:
In general, when there is no overload involved, I prefer to use `static assert` in the body of the function to provide a more friendly error message. Template constraints should only really be used to drive overload / specialization resolution.

Reply:
It's because template constraints are terrible. :>

Reply:
Diagnostics I mean.